examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -0,0 +1,882 @@
+      "{\n",
+      "  float (*restrict f)[f_vec->size[1]] __attribute__ ((aligned (64))) = (float (*)[f_vec->size[1]]) f_vec->data;\n",
+      "  /* Flush denormal numbers to zero in hardware */\n",
+      "  _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);\n",
+      "  _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);\n",
+      "  struct timeval start_section0, end_section0;\n",
+      "  gettimeofday(&start_section0, NULL);\n",
+      "  /* Begin section0 */\n",
+      "  #pragma omp parallel num_threads(nthreads)\n",
+      "  {\n",
+      "    #pragma omp for collapse(1) schedule(static,1)\n",
+      "    for (int x = x_m; x <= x_M; x += 1)\n",
+      "    {\n",
+      "      #pragma omp simd aligned(f:32)\n",
+      "      for (int y = y_m; y <= y_M; y += 1)\n",
+      "      {\n",
+      "        if (f[x + 1][y + 1] > 0)\n",
+      "        {\n",
+      "          f[x + 1][y + 1] = f[x + 1][y + 1] + 1;\n",
+      "        }\n",
+      "      }\n",
+      "    }\n",
+      "  }\n",
+      "  /* End section0 */\n",
+      "  gettimeofday(&end_section0, NULL);\n",
+      "  timers->section0 += (double)(end_section0.tv_sec-start_section0.tv_sec)+(double)(end_section0.tv_usec-start_section0.tv_usec)/1000000;\n",
+      "  return 0;\n",
+      "}\n",
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[2., 0., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 2., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 2., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 2., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 2., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 2., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 2., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 2., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 2., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 0., 2.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "\n",
+    "f.data[:, :] = np.eye(10)\n",
+    "\n",
+    "op = Operator(Eq(f, f + 1, implicit_dims=ci))\n",
+    "print(op.ccode)\n",
+    "op.apply()\n",
+    "\n",
+    "assert (np.count_nonzero(f.data - np.diag(np.diagonal(f.data)))==0)\n",
+    "assert (np.count_nonzero(f.data) == 10)\n",
+    "\n",
+    "f.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now the generated code is as expected and only the elements that were greater than zero were incremented by one.\n",
+    "\n",
+    "\n",
+    "Let's now create a new `Function` `g`, initialized with 1's along its diagonal.\n",
+    "Let's try to add `f` and `g` only if the elements in `g` are not equal to zero and only for the first five rows. Now, we need to satisfy two conditions so we will additionally use SymPy's `And`."
```

Comment:
```
Let's try to add `f` and `g` only if the elements in `g` are not equal to zero and only for the first five rows.
```

would sound better as

```
We want to add `f(x,y)` to `g(x,y)` only if (i) `g != 0` and (ii) `y < 5` (i.e., over the first five columns).
```

in the original sentence there are two flaws:
1) 'only if the elements in `g` are not equal to zero' sounds like that you do the sum only if **all elements in the `g` are ...** , while what you are talking about here is the element-wise sum (that's why I'm using `f(x,y)` and `g(x,y)` in my sentence
2) you say 'rows', but AFAICT you actually mean 'columns' ?

Reply:
```
Now, we need to satisfy ...
```

to be replaced nny

```
To join the two conditions we can use `sympy.And`.
```

Reply:
You are right, fixed